src/frontend/src/components/getApartments.tsx

```diff
@@ -0,0 +1,82 @@
+import { useState, useEffect } from 'react';
+import axios from 'axios';
+
+function getApartments(query: string, pageNum: number, selected: string[]) {
+   const [loading, setLoading] = useState(true);
+   const [error, setError] = useState(false);
+   const array: {
+      // avoids linting warnings
+      name: string;
+      address: string;
+      image: string;
+      review: string;
+      rating: number;
+      price_min: string;
+      price_max: string;
+   }[] = [];
+   const [apartments, setApartments] = useState(array);
+   const [hasMore, setHasMore] = useState(false);
+
+   useEffect(() => {
+      // clears the apartments
+      setApartments(array);
+      pageNum = 1;
+   }, [query, selected]);
+
+   useEffect(() => {
+      // GETs new apartments whenever a button is selected
+      let limit = 2;
+      if (query === '') {
+         limit = 10;
+      }
+      setLoading(true);
+      setError(false);
+      const CancelToken = axios.CancelToken;
+      const source = CancelToken.source();
+      axios({
+         method: 'GET', //http://localhost:3333/mockdata?q=${query}&_page=${pageNum}&_limit=2
+         url: `http://localhost:3333/mockdata?q=${query}&_page=${pageNum}&_limit=${limit}`,
+         cancelToken: source.token,
+      })
+         .then((res) => {
+            const newApartments: {
+               name: string;
+               address: string;
+               image: string;
+               review: string;
+               rating: number;
+               price_min: string;
+               price_max: string;
+            }[] = [];
+            for (let i = 0; i < res.data.length; i++) {
+               if (res.data[i].name !== undefined) {
+                  newApartments.push({
+                     name: res.data[i].name,
+                     address: res.data[i].address,
+                     image: res.data[i].image,
```

Comment:
Similarly here, first thing is that there are multiple images and reviews that will be handled by the right half. Second thing is that the jsons sent by this pull request will not contain them per the Figma design and the Apartments class in the components spreadsheet.

Reply:
> Similarly here, first thing is that there are multiple images and reviews that will be handled by the right half. Second thing is that the jsons sent by the backend for this task will not contain them per the Figma design and the Apartments class in the components spreadsheet.

Reply:
yeah I didn't remove it yet

Reply:
removed